fix(ts): TypeScript-Fehler in Button.tsx und FloatingStars.tsx

## Problem

`npx tsc --noEmit` meldet aktuell **2 TypeScript-Fehler** auf `testing`. CI fängt sie nicht ab, weil der TypeScript-Check in `.github/workflows/ci-cd.yml` mit `|| true` als non-blocking läuft. Beide Fehler sind Pre-existing — sie wurden bei Sprint A (PR #191) und Sprint C (PR #193) bemerkt aber nicht behoben.

## Aktuelle Fehler

### 1. `components/Button.tsx:158` — LinearGradient colors-Tupel

```ts
<LinearGradient
  colors={Colors.gradient.cta as [string, ...string[]]}
  ...
/>
```

```
error TS2769: No overload matches this call.
  Type '[string, ...string[]]' is not assignable to type
  'readonly [ColorValue, ColorValue, ...ColorValue[]]'.
    Source provides no match for required element at position 1 in target.
```

**Ursache:** `expo-linear-gradient` verlangt seit einer SDK-Update ein Tupel mit **mindestens 2** Elementen (`readonly [ColorValue, ColorValue, ...ColorValue[]]`). Der vorhandene Cast `as [string, ...string[]]` reicht nur **1** Element zu — TS sieht keine Garantie für das 2. Pflicht-Element.

**Fix-Optionen:**
- A) Cast anpassen: `as readonly [string, string, ...string[]]` (semantisch korrekt, da `Colors.gradient.cta` immer 2 Werte hat)
- B) `Colors.gradient.*` in `constants/Colors.ts` als `as const`-Tupel typisieren — dann ist kein Cast mehr nötig
- C) Statt Cast: `[Colors.gradient.cta[0], Colors.gradient.cta[1]] as const`

Empfehlung: **B** — typisiert alle Gradient-Verwender automatisch und entfernt den Cast aus allen Call-Sites.

### 2. `components/FloatingStars.tsx:63` — Reanimated style-prop akzeptiert keine Prozent-Strings

```tsx
<Animated.View
  style={[
    styles.item,
    { top: item.top, left: item.left, opacity: item.opacity },  // top/left = '7%' etc.
    animStyle,
  ]}
  ...
/>
```

```
error TS2322: Type '{ top: string; left: string; opacity: number; }' is not assignable …
  Type 'string' is not assignable to type 'DimensionValue | SharedValueDisableContravariance<number> | …
```

**Ursache:** Reanimateds Style-Typing für `Animated.View` ist strikter als React Natives `ViewStyle` — `top`/`left` als Prozent-String wird (zur Zeit) nicht akzeptiert, obwohl es zur Runtime funktioniert. Copilot hat in PR #191 vorgeschlagen, den `as any`-Cast zu entfernen — das war korrekt für RN-Typen, kollidiert aber jetzt mit Reanimated.

**Fix-Optionen:**
- A) Cast wieder hinzufügen: `{ top: item.top as any, left: item.left as any, opacity: item.opacity }` (Pragma: Reanimated-Typing-Bug)
- B) Kurzer Kommentar warum: `// reanimated style typing rejects '%' strings — runtime supports them`
- C) Auf numerische Werte umstellen (item.top/left als `number` × `useWindowDimensions().height`) — mehr Aufwand, aber typesafe

Empfehlung: **A + B** (Cast minimal, mit erklärendem Kommentar).

## Akzeptanzkriterien

- [ ] `npx tsc --noEmit` läuft **fehlerfrei** auf `testing`
- [ ] CI-TypeScript-Check von `|| true` auf **blocking** umgestellt (`.github/workflows/ci-cd.yml`)
- [ ] Keine Regression: `npm test`, `npm run lint` weiter grün
- [ ] Falls Option B (`as const`-Tupel) für #1 gewählt: Alle Call-Sites von `Colors.gradient.*` auf neue Typisierung geprüft

## Bezug

- Bemerkt in PR #191 (Sprint A) und PR #193 (Sprint C), nicht im Scope behoben
- Verwandt: #133 (TypeScript-Fehler in LevelImageDisplay/ThemeContext) — sollte ggf. zusammen mit diesem Issue als "TS-Cleanup"-Sprint abgearbeitet werden

--- a/.github/workflows/ci-cd.yml
+++ b/.github/workflows/ci-cd.yml
@@ -74,7 +74,7 @@ jobs:
         run: npm run validate:svg-counts
 
       - name: TypeScript Type Check
-        run: npx tsc --noEmit || true
+        run: npx tsc --noEmit
 
   # ============================================================================
   # JOB 2: Unit Tests & Coverage

--- a/components/Button.tsx
+++ b/components/Button.tsx
@@ -152,7 +152,7 @@ export function Button({
       >
         {variant === 'primary' && (
           <LinearGradient
-            colors={Colors.gradient.cta as [string, ...string[]]}
+            colors={Colors.gradient.cta}
             start={{ x: 0, y: 0 }}
             end={{ x: 1, y: 0 }}
             style={StyleSheet.absoluteFillObject}

--- a/components/Chip.tsx
+++ b/components/Chip.tsx
@@ -42,12 +42,12 @@ export function Chip({
   }));
 
   const handlePressIn = () => {
-    if (reduceMotion.current) return;
+    if (reduceMotion) return;
     scale.value = withSpring(0.94, { damping: 15, stiffness: 300 });
   };
 
   const handlePressOut = () => {
-    if (reduceMotion.current) return;
+    if (reduceMotion) return;
     scale.value = withSpring(1, { damping: 15, stiffness: 300 });
   };
 

--- a/components/FloatingStars.tsx
+++ b/components/FloatingStars.tsx
@@ -66,7 +66,8 @@ function DecorElement({ item, animate }: { item: DecorItem; animate: boolean }) 
     <Animated.View
       style={[
         styles.item,
-        { top: item.top, left: item.left, opacity: item.opacity },
+        // reanimated style typing rejects '%' strings — runtime supports them
+        { top: item.top as unknown as number, left: item.left as unknown as number, opacity: item.opacity },
         animStyle,
       ]}
       pointerEvents="none"

--- a/components/FloatingStars.tsx
+++ b/components/FloatingStars.tsx
@@ -1,5 +1,5 @@
 import React, { useEffect } from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import { StyleSheet, Text, View, type ViewStyle } from 'react-native';
 import { useReduceMotion } from '../utils/useReduceMotion';
 import Animated, {
   cancelAnimation,
@@ -66,8 +66,8 @@ function DecorElement({ item, animate }: { item: DecorItem; animate: boolean }) 
     <Animated.View
       style={[
         styles.item,
-        // reanimated style typing rejects '%' strings — runtime supports them
-        { top: item.top as unknown as number, left: item.left as unknown as number, opacity: item.opacity },
+        // reanimated style typing rejects '%' strings — cast to ViewStyle['top'] (DimensionValue) preserves semantic intent
+        { top: item.top as ViewStyle['top'], left: item.left as ViewStyle['left'], opacity: item.opacity },
         animStyle,
       ]}
       pointerEvents="none"

--- a/constants/Colors.ts
+++ b/constants/Colors.ts
@@ -15,11 +15,11 @@ export const Colors = {
 
   // Gradient-Kombinationen (für LinearGradient)
   gradient: {
-    primary: ['#7C5CFF', '#F093FB'],     // Primary Gradient (Vivid Purple → Rosa)
-    secondary: ['#F093FB', '#f5576c'],   // Rosa-Gradient
-    warm: ['#FFB547', '#FF6B6B'],        // Apricot-Gradient
-    cta: ['#7C5CFF', '#F093FB'],         // CTA-Gradient mit neuem Primary
-    teal: ['#4ECDC4', '#44CF6C'],        // Teal-Gradient
+    primary:   ['#7C5CFF', '#F093FB'] as const,
+    secondary: ['#F093FB', '#f5576c'] as const,
+    warm:      ['#FFB547', '#FF6B6B'] as const,
+    cta:       ['#7C5CFF', '#F093FB'] as const,
+    teal:      ['#4ECDC4', '#44CF6C'] as const,
   },
 
   // UI Farben - "Warm Paper"

--- a/utils/__tests__/useReduceMotion.test.tsx
+++ b/utils/__tests__/useReduceMotion.test.tsx
@@ -35,7 +35,7 @@ describe('useReduceMotion', () => {
     jest.spyOn(AccessibilityInfo, 'isReduceMotionEnabled').mockResolvedValue(false);
     let capturedHandler: ((rm: boolean) => void) | undefined;
     jest.spyOn(AccessibilityInfo, 'addEventListener').mockImplementation((_event, handler) => {
-      capturedHandler = handler as (rm: boolean) => void;
+      capturedHandler = handler as unknown as (rm: boolean) => void;
       return { remove: jest.fn() } as any;
     });
 

